R/filter_counts.R

```diff
@@ -35,18 +37,39 @@ filter_counts <- function(sce, cr_like = TRUE, fdr_cutoff = 0.01, seed = NULL, .
     stop("cr_like must be set as TRUE or FALSE")
   }
 
+  if(!is.numeric(umi_cutoff) & umi_cutoff >= 0){
+    stop("umi_cutoff must be an integer greater than or equal to 0")
+  }
+
   # calculate probability of being an empty droplet
   if(cr_like){
-    empty_df <- DropletUtils::emptyDropsCellRanger(m = round(counts(sce)), ...)
+    empty_df <- tryCatch(
+      DropletUtils::emptyDropsCellRanger(m = round(counts(sce))),
+      error = function(x){NULL}
+    )
+    metadata(sce)$filtering_method <- "emptyDropsCellRanger"
   } else {
-    empty_df <- DropletUtils::emptyDrops(counts(sce), ...)
+    empty_df <- tryCatch(
+      DropletUtils::emptyDrops(counts(sce), ...),
+      error = function(x){NULL}
+    )
+    metadata(sce)$filtering_method <- "emptyDrops"
```

Comment:
I don't love the repetition here. I have some ideas for how to clean this up, but I can't do it as a comment. I will file a PR to this branch with my ideas.